actionpack/lib/action_dispatch/middleware/remote_ip.rb

```diff
@@ -116,10 +116,18 @@ def calculate_ip
         forwarded_ips = ips_from(@req.x_forwarded_for).reverse
 
         # +Client-Ip+ and +X-Forwarded-For+ should not, generally, both be set.
-        # If they are both set, it means that this request passed through two
-        # proxies with incompatible IP header conventions, and there is no way
-        # for us to determine which header is the right one after the fact.
-        # Since we have no idea, we give up and explode.
+        # If they are both set, it means that either:
+        #
+        # 1) this request passed through two proxies with incompatible IP header
```

Comment:
Capitalize each number list entry. i.e. "This" instead of "this". Don't end with an "or". They are separate list entries and should stand on their own. End each list entry in a period.